src/canvas/guard/canvas.public.guard.ts

```diff
@@ -24,6 +25,10 @@ export class CanvasPublicGuard implements CanActivate {
     const canvasAccess = canvasGuard.canActivate(context);
 
     const request = context.switchToHttp().getRequest();
+    if (!uuidValidate(request.params.id)) {
+      return true;
```

Comment:
Are you sure that it should return true when uuid is not valid? Or am I missing the point somewhere? 😀

Reply:
Yes. It has nothing to do with authentication. Validator should throw exception

Reply:
As this is called before validation. It would fail here when it's not uuid, so I pass it here